parser/proto.y

```diff
@@ -327,269 +334,231 @@ stringLit : _STRING_LIT {
 		$$ = &stringList{$1, $2}
 	}
 
-aggregate : '{' aggFields '}' {
+messageLiteralWithBraces : '{' messageTextFormat '}' {
 		fields, delims := $2.toNodes()
 		$$ = ast.NewMessageLiteralNode($1, fields, delims, $3)
 	}
 	| '{' error '}' {
 	    $$ = nil
 	}
 
-aggFields : aggField {
-		if $1 != nil {
-			$$ = &messageFieldList{$1, nil}
-		} else {
-			$$ = nil
-		}
-	}
-	| aggField aggFields {
-		if $1 != nil {
-			$$ = &messageFieldList{$1, $2}
-		} else {
-			$$ = $2
-		}
-	}
-	| {
-		$$ = nil
+messageTextFormat : messageLiteralFields {
+		$$ = $1
 	}
 
-aggField : aggFieldEntry {
+messageLiteralFields : messageLiteralField ',' messageLiteralFields {
 		if $1 != nil {
-			$$ = &messageFieldEntry{$1, nil}
+			entry := &messageFieldEntry{$1, $2}
+			$$ = &messageFieldList{entry, $3}
 		} else {
 			$$ = nil
 		}
 	}
-	| aggFieldEntry ',' {
+	| messageLiteralField ';' messageLiteralFields {
 		if $1 != nil {
-			$$ = &messageFieldEntry{$1, $2}
-		} else {
-			$$ = nil
-		}
+			entry := &messageFieldEntry{$1, $2}
+                	$$ = &messageFieldList{entry, $3}
+                } else {
+                	$$ = nil
+       		}
 	}
-	| aggFieldEntry ';' {
+	| messageLiteralField messageLiteralFields {
 		if $1 != nil {
-			$$ = &messageFieldEntry{$1, $2}
+			entry := &messageFieldEntry{$1, nil}
+                	$$ = &messageFieldList{entry, $2}
 		} else {
 			$$ = nil
 		}
 	}
-	| error ',' {
-		$$ = nil
-	}
-	| error ';' {
-		$$ = nil
-	}
-	| error {
-		$$ = nil
-	}
-
-aggFieldEntry : aggName ':' scalarConstant {
+	| messageLiteralField ',' {
 		if $1 != nil {
-			$$ = ast.NewMessageFieldNode($1, $2, $3)
+			entry := &messageFieldEntry{$1, $2}
+                	$$ = &messageFieldList{entry, nil}
 		} else {
 			$$ = nil
 		}
 	}
-	| aggName '[' ']' {
+	| messageLiteralField ';' {
 		if $1 != nil {
-			val := ast.NewArrayLiteralNode($2, nil, nil, $3)
-			$$ = ast.NewMessageFieldNode($1, nil, val)
+			entry := &messageFieldEntry{$1, $2}
+			$$ = &messageFieldList{entry, nil}
 		} else {
 			$$ = nil
 		}
 	}
-	| aggName ':' '[' ']' {
+	| messageLiteralField {
 		if $1 != nil {
-			val := ast.NewArrayLiteralNode($3, nil, nil, $4)
-			$$ = ast.NewMessageFieldNode($1, $2, val)
+			entry := &messageFieldEntry{$1, nil}
+			$$ = &messageFieldList{entry, nil}
 		} else {
 			$$ = nil
 		}
 	}
-	| aggName '[' msgList ']' {
-		if $1 != nil {
-			vals, commas := $3.toNodes()
-			val := ast.NewArrayLiteralNode($2, vals, commas, $4)
-			$$ = ast.NewMessageFieldNode($1, nil, val)
-		} else {
-			$$ = nil
-		}
-	}
-	| aggName ':' '[' constantList ']' {
-		if $1 != nil {
-			vals, commas := $4.toNodes()
-			val := ast.NewArrayLiteralNode($3, vals, commas, $5)
-			$$ = ast.NewMessageFieldNode($1, $2, val)
-		} else {
-			$$ = nil
-		}
-	}
-	| aggName ':' '[' error ']' {
-		$$ = nil
-	}
-	| aggName '[' error ']' {
+	| {
 		$$ = nil
 	}
```

Comment:
This is a bit confusing looking. But after reading through this all, I think I understand: this production can literally be empty. Maybe a comment to that effect? Or add explicit empty productions for `'{' '}'` and `'<' '>'` to `messageLiteralWithBraces` and `messageLiteral` productions?